user_manager/models.py

```diff
@@ -0,0 +1,87 @@
+"""
+Models for User Manager Application.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.db import models
+
+
+class UserManagerRole(models.Model):
+    """
+    Creates a manager-managee link between users.
+
+    .. note::
+        In case the manager doesn't have an account registered in the system,
+        their email will be linked instead, and auto-upgraded to a foreign key
+        when they register an account.
+    """
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        related_name='+',
+    )
+    manager_user = models.ForeignKey(
+        User,
+        null=True,
+        blank=True,
+        on_delete=models.CASCADE,
+        related_name='+',
+    )
+    # This will get upgraded to a foreign key to the manager's user account
+    # when they register.
+    unregistered_manager_email = models.EmailField(
+        null=True,
```

Comment:
Django docs say to "Avoid using null on string-based fields": https://docs.djangoproject.com/en/2.0/ref/models/fields/#null

Reply:
I remember why I added the null now. It's needed for the unique together constraint:
```python
unique_together = (
            ('user', 'manager_user'),
            ('user', 'unregistered_manager_email'),
        )
```
Without ``null=True`` this constraint fille fail because there will be many pairs the same user with a blank email.

Reply:
Oh, that makes sense.

Reply:
> I remember why I added the null now. It's needed for the unique together constraint:

Can you please just add that as a comment on the line here where we define `null=True` ?